src/main/webapp/src/sessions/gapi.session.ts

```diff
@@ -0,0 +1,217 @@
+   */
+  get isSignedIn(): boolean {
+    return this.auth2.isSignedIn.get();
+  }
+
+  /**
+   * Returns whether the user has accepted the consent prompt in the current session
+   */
+  get consent(): boolean {
+    return this.hasConsent;
+  }
+
+  /**
+   * Google Authentication; displays consent prompt for Google Calendar API
+   */
+  signIn() {
+    return this.auth2
+      .signIn({
+        prompt: 'consent',
+      })
+      .then(() => {
+        this.hasConsent = true;
+      });
+  }
+
+  /**
+   * Sign out from Google Auth
+   */
+  signOut(): void {
+    this.auth2.signOut();
+  }
+
+  /**
+   * Retrieve calendar events given a date and call function to display the events in the page
+   * @param selectedDate - start date and time (min)
+   * @param selectedDateMax - end of the day of selected date (max)
+   */
+  listEvents(
+    gardenName: string,
+    selectedDate: string,
+    selectedDateMax: string
+  ) {
+    gapi.client.load('calendar', 'v3', () => {
+      gapi.client.calendar.events
+        .list({
+          calendarId: 'primary',
+          timeMin: selectedDate,
+          timeMax: selectedDateMax,
+          singleEvents: true,
+          orderBy: 'startTime',
+        })
+        .then(response => {
+          const events = response.result.items;
+          this.tasks.clearTasks();
+          if (events.length > 0) {
+            for (let i = 0; i < events.length; i++) {
+              const event = events[i];
+              if (event.summary.indexOf(gardenName) !== -1) {
+                const eventTime = this.getEventTime(event.start.dateTime);
+                this.tasks.createTaskElement(
+                  eventTime,
+                  event.summary,
+                  event.attendees,
+                  event.description
+                );
+              }
+            }
+          }
+        });
+    });
+  }
+
+  /**
+   * Get the start time of the event otherwise return an All Day string
+   * @param dateTime: string containing date and time
+   */
+  getEventTime(dateTime: string): string {
+    if (dateTime) {
+      const dateTimeSplit = dateTime.split('T');
+      const time = dateTimeSplit[1].split('-', 1);
+      return time[0];
+    } else {
+      return 'All Day';
+    }
+  }
+
+  /**
+   * Inserts a new calendar event into the users' primary calendar
+   * @param title - name of the event (known as summary in the calendar api)
+   * @param startDateTime - start date and time of the event in ISO format
+   * @param endDateTime - end date and time of the event in ISO format
+   * @param timezone - set the timezone for the time
+   * @param participants - members invited to event
+   * @param description - (optional) description of the event
+   */
+  createEvent(
+    title: string,
+    startDateTime: string,
+    endDateTime: string,
+    timezone: string,
+    participants: string[],
+    description?: string
+  ) {
+    if (typeof description === 'undefined') {
```

Comment:
You can simplify this as `description = description ?? ''` or `description = description || ''` (if you wish to guard not only against null and undefined but also any falsy value)

Reply:
Awesome! thank you